Recommendations: Add event tracking for Podroll cells + Info modal

bjtitus opened the pull request:
| 📘 Part of: #2989 |
|:---:|

Adds events for tracking in the Podroll section.

## To test

1. Apply [this patch](https://github.com/user-attachments/files/20019416/StaticRecommendationsChange.patch) to point to a static example since the API isn't ready yet.
2. Go to a podcast page
3. Tap on the "Similar Shows" tab
4. Tap on the Podroll header "Recommended shows by the creator"
5. ✅ Verify the Tracks event `podcast_screen_podroll_information_model_shown` is logged
6. Tap on a podcast subscribe button
7. ✅ Verify the Tracks event `podcast_screen_podroll_podcast_subscribed` is logged with the property `podcast_uuid`
8. Tap on a podcast row to open the podcast page
9. ✅ Verify the Tracks event `podcast_screen_podroll_podcast_tapped` is logged with the property `podcast_uuid`

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

bjtitus committed:
Add event tracking for Podroll cells + Info modal

dangermattic commented:
<!--
  0 Errors
  1 Warning: View files have been modified,...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

danielebogo commented on podcasts/PodcastViewController+TableData.swift:
```diff
@@ -185,15 +185,22 @@ extension PodcastViewController: UITableViewDataSource, UITableViewDelegate {
                 let cell = tableView.dequeueReusableCell(withIdentifier: PodcastTableViewCell.reuseIdentifier, for: indexPath) as! PodcastTableViewCell
-                cell.configure(with: podcast, datetime: nil)
+                cell.configure(with: podcast, datetime: nil) { viewModel in
+                    var properties = ["podcast_uuid": viewModel.uuid]
```
Should this be `let`?

danielebogo reviewed (changes requested):
@bjtitus I see all events but `podcastScreenPodrollPodcastSubscribed` is never tracked. I also set a breakpoint into the block but it's never called

bjtitus committed:
Change analytics from similar shows to you might like

bjtitus committed:
Fix logging podroll subscribed events

bjtitus commented:
@danielebogo Woops, sorry. I've updated with a fix and a change to rename Similar Shows to You Might Like. I rebased on the latest from the upstream branch so had to force push.

You should see two events based on the section for the "You might like" tab.

Tapping in the top podroll section:
```
🔵 Tracked: podcast_subscribed ["theme_light_preference": "default_light", "theme_dark_preference": "default_dark", "source": "podcast_screen_you_might_like", "uuid": "e6d1eff0-244e-012e-04d4-00163e1b201c", "theme_selected": "default_light", "theme_use_system_settings": true]

🔵 Tracked: podcast_screen_podroll_podcast_subscribed ["podcast_uuid": "e6d1eff0-244e-012e-04d4-00163e1b201c", "theme_selected": "default_light", "theme_use_system_settings": true, "theme_dark_preference": "default_dark", "theme_light_preference": "default_light"]
```

Tapping in the bottom related shows:
```
🔵 Tracked: podcast_subscribed ["theme_light_preference": "default_light", "source": "podcast_screen_you_might_like", "uuid": "045f3aa0-aa6a-0136-7b93-27f978dac4db", "theme_selected": "default_light", "theme_use_system_settings": true, "theme_dark_preference": "default_dark"]

🔵 Tracked: podcast_screen_you_might_like_subscribed ["theme_dark_preference": "default_dark", "theme_light_preference": "default_light", "podcast_uuid": "045f3aa0-aa6a-0136-7b93-27f978dac4db", "theme_selected": "default_light", "list_datetime": "2025-04-08T01:40:01Z", "theme_use_system_settings": true]
```

bjtitus committed:
Change properties to immutable

danielebogo reviewed (approved):
Thanks @bjtitus ! All good now!

bjtitus committed:
Merge branch 'trunk' into bjtitus/recommendations/tracks-events-podroll

# Conflicts:
#	podcasts/Analytics/Helpers/AnalyticsCoordinator.swift
#	podcasts/PodcastTableCellView.swift
